### PR TITLE
Problem: recent optimizations not included

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- add several optimizations in dependencies: a) async fireEvents, b) faster prepare proposal when using NopMempool, c) parallel check-tx

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

## Release Notes

- **New Features**
  - Added support for registering payee and counterpartyPayee in the relayer precompile.

- **Improvements**
  - Enhanced performance for asynchronous event firing and proposal preparation.
  - Updated dependencies to improve compatibility and performance.
  - Introduced a testnet benchmark command for better performance evaluation.
  - Implemented parallel generation of test transactions and improved load generator behavior.
  - Enhanced transaction validation process with byte data for more robust handling.

- **Bug Fixes**
  - Resolved issues with single validator benchmarks and ensured proper node shutdown.
  - Fixed state overwrites in debug trace APIs and improved transaction validation logic.
  - Addressed issues with multisig accounts and unsupported sign modes.

These changes enhance the application's functionality and performance, providing a more reliable user experience.

### CODE REVIEW

**Review by mmsqe — APPROVED** [review]
https://github.com/cometbft/cometbft/compare/v0.38.13...crypto-org-chain:cometbft:f2b043f6eebe
https://github.com/crypto-org-chain/cosmos-sdk/compare/d78d66e74712...a70c19a03a81
https://github.com/crypto-org-chain/ethermint/compare/e567c5ad3acf...8afdc6924206

File: app/proposal.go [hunk 1]
```diff
@@ -55,6 +47,31 @@
...
+				continue
+			}
+			filtered = append(filtered, txBz)
+		}
+
+		txs = filtered
+	}
+
+	return ts.TxSelector.SelectTxForProposalFast(ctx, txs)
+}
```
[the remark this answers is not in the document]
Comment (yihuang) [reply app/proposal.go:73]: yes, try it

[the remark this answers is not in the document]
Comment (yihuang) [reply app/proposal.go:73]: very good 👍

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@
 * [#1664](https://github.com/crypto-org-chain/cronos/pull/1664) Update cometbft to 0.38.13.
 * [#1660](https://github.com/crypto-org-chain/cronos/pull/1660) Support async check tx.
 * [#1667](https://github.com/crypto-org-chain/cronos/pull/1667) Add testnet benchmark command.
+* [#1669](https://github.com/crypto-org-chain/cronos/pull/1669) Add performance optimizations: a) async fireEvents, b) faster prepare proposal when using NopMempool
```
Comment (yihuang) [inline CHANGELOG.md:14]:
```suggestion
* [#1669](https://github.com/crypto-org-chain/cronos/pull/1669) Add performance optimizations: a) async fireEvents, b) faster prepare proposal when using NopMempool, c) parallel check-tx
```

**Pull request merged**